lib/src/route_data.dart

```diff
@@ -0,0 +1,87 @@
+// ignore_for_file: public_member_api_docs
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:meta/meta_meta.dart';
+
+import 'go_route.dart';
+import 'go_router_state.dart';
+
+/// Baseclass for types that wish to play along her.
+///
+/// This looks like [StatelessWidget] – I don't think we want to subclass,
+/// though.
+abstract class GoRouteData {
+  const GoRouteData();
+
+  Widget build(BuildContext context) =>
+      throw UnsupportedError('Should be overridden in subclass.');
+
+  Page<dynamic> createPage(
+    BuildContext context,
+    GoRouterState state,
+  ) {
+    if (context.findAncestorWidgetOfExactType<CupertinoApp>() != null) {
```

Comment:
@csells – this feels the least wrong of any impl I could come up with. Thoughts?